kqueen/blueprints/api/views.py

```diff
@@ -220,6 +220,29 @@ def cluster_resize(pk):
     return jsonify(output)
 
 
+@api.route('/clusters/<uuid:pk>/set_network_policy', methods=['PATCH'])
+@jwt_required()
+def cluster_set_network_policy(pk):
+    obj = get_object(Cluster, pk, current_identity)
+
+    data = request.json
+    if not isinstance(data, dict) or (isinstance(data, dict) and 'network_policy' not in data):
+        msg = 'Failed to get network policy configuration'
+        logger.error(msg)
+        abort(400, description=msg)
+
+    res_status, res_msg = obj.engine.set_network_policy(data['network_policy']['provider'],
```

Comment:
Adding of `network_policy` key makes sense only if you pass it with cluster creation. Otherwise it is redundant as you send a request to `set_network_policy`

Reply:
By the way, could you add an ability to set policy with cluster creation? It's easier when you do only one request

Reply:
no, u in several cases u can manage policy on-the-fly

Reply:
or i don't understand u , pls clarify info